tests/validate_perfetto.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+import sys
+import argparse
+from perfetto.trace_processor import TraceProcessor
+
+def validate_perfetto(labels, counts, depths):
+    expected = []
+    for litr, citr, ditr in zip(labels, counts, depths):
+        entry = []
+        _label = litr
+        if ditr > 0:
+            _label = "{}".format(litr)
+        entry = [_label, citr, ditr]
+        expected.append(entry)
+
+    for ditr, eitr in zip(perfetto_data, expected):
+        _label = ditr["label"]
+        _count = ditr["count"]
+        _depth = ditr["depth"]
+        
+        if _label != eitr[0]:
+            raise RuntimeError(f"Mismatched prefix: {_label} vs. {eitr[0]}")
+        if _count != eitr[1]:
+            raise RuntimeError(f"Mismatched count: {_count} vs. {eitr[1]}")
+        if _depth != eitr[2]:
+            raise RuntimeError(f"Mismatched depth: {_depth} vs. {eitr[2]}")
+
+
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    
+    parser.add_argument(
+        "-l", "--labels", nargs="+", type=str, help="Expected labels", default=[]
+    )
+    parser.add_argument(
+        "-c", "--counts", nargs="+", type=int, help="Expected counts", default=[]
+    )
+    parser.add_argument(
+        "-d", "--depths", nargs="+", type=int, help="Expected depths", default=[]
+    )
+    parser.add_argument("-i", "--input", type=str, help="Input file", required=True)
+
+    args = parser.parse_args()
+
+    if len(args.labels) != len(args.counts) or len(args.labels) != len(args.depths):
+        raise RuntimeError(
+            "The same number of labels, counts, and depths must be specified"
+        )
+
+
+tp = TraceProcessor(trace=(args.input))

```

Comment:
Everything below here should be indented inside the `if __name__ == "__main__":` block.